tests/environment.py

```diff
@@ -336,3 +349,83 @@ def launch_environment_docker(unique_id: str, skip_init: bool = False, tag: str
     debug('agent2: {}'.format(agents[1].mac))
     debug('agent2 wlan0: {}'.format(agents[1].radios[0].mac))
     debug('agent2 wlan2: {}'.format(agents[1].radios[1].mac))
+
+
+class ALEntityPrplWrt(ALEntity):
+    """Abstraction of ALEntity in real device."""
+
+    def __init__(self, name: str, device: None, is_controller: bool = False):
+        self.device = device
+        self.name = name
+        self.bridge_name = 'br-lan'
+
+        if is_controller:
+            self.config_file_name = '/opt/prplmesh/config/beerocks_controller.conf'
+        else:
+            self.config_file_name = '/opt/prplmesh/config/beerocks_agent.conf'
+
+        ucc_port = self.command(("grep \"ucc_listener_port\" {} "
+                                "| cut -d'=' -f2 | cut -d\" \" -f 1").format(self.config_file_name))
+
+        device_ip_output = self.command(
+                "ip -f inet addr show {} | head -n 2".format(self.bridge_name))
+        device_ip = re.search(
+            r'inet (?P<ip>[0-9.]+)', device_ip_output.decode('utf-8')).group('ip')
+        self.log_folder = self.command(
+            "grep log_files_path {} | cut -d=\'=\' -f2".format(self.config_file_name))
+
+        ucc_socket = UCCSocket(device_ip, ucc_port)
+        mac = ucc_socket.dev_get_parameter('ALid')
+
+        super().__init__(mac, ucc_socket, installdir, is_controller)
+
+        # We always have two radios, wlan0 and wlan2
+        RadioHostapd(self, "wlan0", device=self)
+        RadioHostapd(self, "wlan2", device=self)
+
+    def command(self, *command: str) -> bytes:
+        """Execute `command` in device and return its output."""
+        self.device.sendline(command)
+        return self.device.read()
+
+    def wait_for_log(self, regex: str, start_line: int, timeout: float) -> bool:
```

Comment:
I searched for the means that I have, as a user of this function, to supply a value for `start_line` - and i couldn't find.
I don't think that there is a way for the user to get a value for it. I also looked at [test_flow.py](https://github.com/prplfoundation/prplMesh/blob/feature/n1072-prplwrt-device/tests/test_flows.py) and couldn't find a use of it.
Normally the interface exposes a way to know where to start the next search, e.g. by returning the last position it was found, or by explicit function call `get_last_searched_line()`

Reply:
> Normally the interface exposes a way to know where to start the next search, e.g. by returning the last position it was found

[That's exactly what it does.](https://github.com/prplfoundation/prplMesh/blob/master/tests/environment.py#L187) However, this was never documented because I consider it a horrible interface, and it should be converted to use a [checkpoint mechanism](https://github.com/prplfoundation/prplMesh/blob/master/tests/environment.py#L155).